### PR TITLE
LayoutSinglePostWeek31

### PR DESCRIPTION
**Pull request by mrnschrmm**

### CODE REVIEW

File: Components/BlockPostHeader/Partials/Image/_style.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.image {
+  display: block;
+  height: auto;
+  margin: 0;
+  width: 100%;
```
Comment (steffenbew) [inline Components/BlockPostHeader/Partials/Image/_style.scss]: I think you don't need all of the above. Instead though, please use the same markup and styling as in `BlockImage` (with <figure>).

File: Components/BlockPostHeader/Partials/Image/index.twig [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<div class="image">
+  <img class="image-tag lazyload"
+    src="{{ image.src|resizeDynamic(840) }}"
+    srcset="{{ placeholderImage(840, 840 / image.aspect) }}"
+    data-srcset="
+      {{ image.src|resizeDynamic(840) }} 840w,
```
Comment (steffenbew) [inline Components/BlockPostHeader/Partials/Image/index.twig]: Please prepare also high resolution sizes.

File: Components/BlockPostHeader/_style.scss [hunk 3]
```diff
@@ -0,0 +1,109 @@
+[is='flynt-block-post-header'] {
+
+  // ##BLOCK-POST-HEADER
+  // =============================================================
+
+  // SPACING
+  // -------------------------------------------------------------
+
+  .container {
+    @include center;
```
Comment (steffenbew) [inline Components/BlockPostHeader/_style.scss]: Instead, please use the helper class in the markup directly.

File: Components/BlockPostHeader/_style.scss [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      margin-right: 0.25rem;
+    }
+
+    &-readingtime {
+      margin-left: 0.25rem;
+    }
+
+    &-intro {
+      p {
+        font-size: 1.125rem;
```
Comment (steffenbew) [inline Components/BlockPostHeader/_style.scss]: I would prefer using one of our headline classes instead, like at the top in the BaseStyle.

File: Components/BlockPostHeader/index.twig [hunk 5]
```diff
@@ -0,0 +1,25 @@
+<div class="flyntComponent" is="flynt-block-post-header">
+  <div class="container">
+    <div class="blockPostHeader">
+      <div class="blockPostHeader-title">
+        <h1 class="blockPostHeader-h1">{{ post.title }}</h1>
+      </div>
+      <div class="blockPostHeader-meta">
+        {% if post.date %}
+          <time class="blockPostHeader-date">{{ post.date|date }}</time> {{ generalTranslatable.authorDivider|e }}
```
Comment (steffenbew) [inline Components/BlockPostHeader/index.twig]: I think it's missing the timestamp here.

File: Components/BlockPostHeader/index.twig [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        {% endif %}
+        {{ generalTranslatable.postedByLabel|e }} <a class="blockPostHeader-author" href="{{ post.author.link|e('esc_url') }}">{{ post.author.name|capitalize }}</a>
```
Comment (steffenbew) [inline Components/BlockPostHeader/index.twig]: Not sure we should capitalize. I feel it should be the actual nick name.

File: Components/BlockPostHeader/index.twig [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        {{ generalTranslatable.postedInLabel|e }} <a class="blockPostHeader-category" href="{{ post.category.link|e('esc_url') }}">{{ post.category }}</a>
+        {{ generalTranslatable.readingtimeDivider|e }} <span class="blockPostHeader-readingtime">{{ post.content|readingtime }} {{ generalTranslatable.readingtimeLabel|e }}</span>
+      </div>
+      <div class="blockPostHeader-intro">
+        <p>{{ post.preview.read_more(false) }}</p>
```
Comment (steffenbew) [inline Components/BlockPostHeader/index.twig]: Does this work now also with not having a manual excerpt, or does it show the beginning of the actual post content instead?

File: Components/BlockPostHeader/index.twig [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+      </div>
+      {% if post.thumbnail %}
+        <div class="blockPostHeader-media">
+          {% include 'Partials/Image/index.twig' with {image: post.thumbnail} only %}
```
Comment (steffenbew) [inline Components/BlockPostHeader/index.twig]: Couldn't we also render the image component here and pass the image as an option as well?

File: assets/styles/_base.scss [hunk 9]
```diff
@@ -135,6 +135,45 @@ body {
...
+
+      @media (min-width: $breakpoint-tablet) {
+        margin-top: 3rem;
+      }
+    }
+
+    &-main,
+    &-related {
+      > .flyntComponent {
+        margin: 0;
```
Comment (steffenbew) [inline assets/styles/_base.scss]: This shouldn't be the case. At most the first and last components margin-/padding-top / -bottom.

File: Components/BlockPostFooter/_style.scss [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+      &:not(:last-child) {
+        margin: 0 8px 0 0;
+
+        @media (min-width: $breakpoint-tablet) {
+          margin-right: 12px;
+        }
+      }
+    }
+
+    &-tagLink {
```
[the remark this answers is not in the document]
Comment (mrnschrmm) [reply Components/BlockPostFooter/_style.scss]: I left what is still necessary to let it look like before or like in GridPostArchive. I assume the pill class isn't complete yet.

File: Components/BlockPostHeader/Partials/Image/_style.scss (hunk 1 above)
Comment (mrnschrmm) [reply Components/BlockPostHeader/Partials/Image/_style.scss]: Integrated BlockImage ... but generally there is no CSS at all related to the image or structures shown in the lazysizes docs.

**Pull request merged**